os_stub/memlib/copy_mem.c

```diff
@@ -9,35 +9,65 @@
  **/
 
 #include "base.h"
+#include "library/debuglib.h"
 
 /**
- * Copies a source buffer to a destination buffer, and returns the destination buffer.
+ * Copies bytes from a source buffer to a destination buffer.
  *
- * This function copies length bytes from source_buffer to destination_buffer, and returns
- * destination_buffer.  The implementation must be reentrant, and it must handle the case
- * where source_buffer overlaps destination_buffer.
+ * This function copies "len" bytes from "src_buf" to "dst_buf".
  *
- * If length is greater than (MAX_ADDRESS - destination_buffer + 1), then ASSERT().
- * If length is greater than (MAX_ADDRESS - source_buffer + 1), then ASSERT().
+ * Asserts and returns a non-zero value if any of the following are true:
+ *   1) ("src_buf" or "dst_buf" are NULL) and "len" is greater 0.
```

Comment:
for 1), I recommend to remove "len is greater 0.". the src_buf and dst_buf shall not be NULL in all cases.

Reply:
One of the test cases (test_spdm_responder_challenge_auth_case1) fails if I make this change. I'll investigate.

Reply:
A few places call copy_mem with null source and size 0. I'll push the code that corrects it for those places, caught by current unit tests.

Reply:
Thank you. I see you updated the code. Would you please update comment as well?

Reply:
I think I updated the copy function comments over a couple different change sets. If you look at the final changes, they should be in there?

Reply:
"len is greater 0" is still there in the final, which is not expected. Would you please double check?